PolkAuction milestone 2

CrommVardek opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/452

takahser commented:
Hi @CrommVardek 
Thanks for the delivery!
I saw that you decided to skip the _2.b. | Parachain page_ deliverable. Hence we require you to [amend your original grant application](https://github.com/w3f/Grants-Program#changes-to-a-grant-after-approval). Since all the information is already contained in other pages you indeed did deliver I think it's not necessary to update the costs as part of the amendment. 👍

CrommVardek commented:
Hi @takahser ,
Thanks for your first feedback, I've updated the application ( https://github.com/w3f/Grants-Program/pull/818 ) !

takahser commented:
Hi @CrommVardek,
Thanks for your patience!
You can find my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/polk-auction_2_takahser.md).
Please note that I changed the IPs in the core repo to 127.0.0.1, see in [my fork of your polk-auction-core repo](https://github.com/takahser/polk-auction-core/commit/6a715b388675ab6030694a1840ca921dd78d67a2) since the docker containers wouldn't start otherwise.

takahser commented:
Hi @CrommVardek 

> Did you used npm to install the dependencies or yarn ?

I used yarn, as described in your README.

>  can you tell me what's the docker version you used, on what OS you were running docker (shouldn't matter, but who knows)

Sorry, I forgot to mention this:
I'm using a M1-chipped macbook on macOS 12.2.

Docker:
```bash
% docker --version
Docker version 20.10.11, build dea9396
```

Let me know if you need more information or have further questions.

CrommVardek commented:
Hi @takahser, 

I've resolved some things, but not everything, I'll come back to you once everything is fixed & re-tested on my side.

takahser commented:
Hi @CrommVardek 
Alright, thanks for the update. Looking forward!

CrommVardek commented:
Hi @takahser, 

So I've resolved the issue regarding the docker build (and indeed, the command docker-compose works only if the previous build command works). This was due to the new parcel version. I've downgraded it.

I've also implemented the suggestion regarding the header's relay chain selection. 

I've fix the warnings related to key of list's elements.

I've added a FAQ section in the about page.

(Not in the evaluation) : I've fixed a bug when data for a selected chain was loading (pending request) and the user switched chain, the data displayed were from the previously selected chain. So it is now fixed. The request also now use a cache mecanism because : parachains, auction and crowdloan are not changing too much over short periods of time and because the queries (especially for crowdloan) were taking several seconds, which can be annoying.

Finally, I cannot replicate the other two errors `parachains.filter` and `Cannot read properties of undefined (reading 'filter')`. I think that there might be some misconfiguration with the backend. Can you check that polk-auction core is running (for example accessing `http://127.0.100.1:8080/parachain/kusama` should return data) and that the public nodes are used (that you did not change the `SAS_SUBSTRATE_WS_URL` in both docker-compose files in polk-auction-core) ? 

You can pull the head of the default branches for both repositories to continue your evaluation.

CrommVardek commented:
Hello, just checking if everything is alright with my last commits regarding the evaluation ?

Thanks !

takahser commented:
@CrommVardek apologies, it's been quite busy recently which is why I didn't find the time to update my evaluation yet.
Currently, I have no open questions, I'm going to provide feedback as soon as I can.

CrommVardek commented:
@takahser : Thanks, no problem, was just checking my PR was still "alive", take your time to evaluate and provide feedback.

takahser commented:
@CrommVardek I just checked out your latest commit and tried to run the containers again.
I'm now finally able to
- build the image: `docker build -t polk-auction-ui .`
- and start it: `docker-compose -f "docker/polkauction-core/docker-compose.yml" up -d --build` 

```bash
$ docker ps
CONTAINER ID   IMAGE                                 COMMAND                  CREATED         STATUS         PORTS                                                                                                                             NAMES
87e173464949   polk-auction-core:latest              "java -Djava.securit…"   6 seconds ago   Up 5 seconds   3308/tcp, 127.0.100.1:8080->8080/tcp                                                                                              polk-auction-core
```

So far, so good.
However, for some reason, I'm not able to access the api on `:8080` on the host machine.

Also, it's not clear to me if I need to start the Polkadot or Kusama Node and sidecar API along with it. Feel free to update this in the README.

CrommVardek commented:
Hi @takahser ,

Regarding the port, is it possible you have another running app that is using the 8080 port ?

I indeed realize the README is unclear. I'll update it, in the mean-time, given the current configuration, you need to run only the sidecar API (for both Polkadot and Kusama) - the nodes used are the public endpoint provided by Parity.

CrommVardek commented:
Hi @takahser , I've updated the [README](https://github.com/CrommVardek/polk-auction-core/blob/develop/README.md)

I hope it's more clear, but for your usage, you can run sidecars this way : 

`docker-compose -f "docker\polkadot\docker-compose.yml" up -d --build sidecar`
 
and 

`docker-compose -f "docker\kusama\docker-compose.yml" up -d --build sidecar`

takahser commented:
@CrommVardek thank you, that sounds good. I'll get back to you soon, thanks again for your patience!

takahser commented:
@CrommVardek 

Thanks for your patience and apologies for the delay. Also thanks for clarifying the README.

I checked out your current code and still found some issues.

1. While the frontend tests succeed now, the backend tests fail:


    ```bash
    $ mvn test
    WARNING: An illegal reflective access operation has occurred
    WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
    WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
    WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
    WARNING: All illegal access operations will be denied in a future release
    [INFO] Scanning for projects...
    [WARNING] 
    [WARNING] Some problems were encountered while building the effective model for vardek:polk-auction.core:jar:0.2.0-BETA
    [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 207, column 21
    [WARNING] 
    [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
    [WARNING] 
    [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
    [WARNING] 
    [INFO] 
    [INFO] ----------------------< vardek:polk-auction.core >----------------------
    [INFO] Building polk-auction.core 0.2.0-BETA
    [INFO] --------------------------------[ jar ]---------------------------------
    [INFO] 
    [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ polk-auction.core ---
    [INFO] Using 'UTF-8' encoding to copy filtered resources.
    [INFO] Copying 6 resources
    [INFO] 
    [INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ polk-auction.core ---
    [INFO] Nothing to compile - all classes are up to date
    [INFO] 
    [INFO] --- kotlin-maven-plugin:1.4.32:compile (compile) @ polk-auction.core ---
    [WARNING] Using experimental Kotlin incremental compilation
    [INFO] Compiled 0 Kotlin files using incremental compiler
    [INFO] 
    [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ polk-auction.core ---
    [INFO] Using 'UTF-8' encoding to copy filtered resources.
    [INFO] Copying 4 resources
    [INFO] 
    [INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ polk-auction.core ---
    [INFO] Nothing to compile - all classes are up to date
    [INFO] 
    [INFO] --- kotlin-maven-plugin:1.4.32:test-compile (test-compile) @ polk-auction.core ---
    [WARNING] Using experimental Kotlin incremental compilation
    [INFO] Compiled 0 Kotlin files using incremental compiler
    [INFO] 
    [INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ polk-auction.core ---
    [INFO] 
    [INFO] -------------------------------------------------------
    [INFO]  T E S T S
    [INFO] -------------------------------------------------------
    [ERROR] Mar 11, 2022 6:21:29 PM org.junit.platform.launcher.core.LauncherConfigurationParameters loadClasspathResource
    [ERROR] INFO: Loading JUnit Platform configuration parameters from classpath resource [file:/home/seraya/repos/polk-auction/polk-auction-core/target/test-classes/junit-platform.properties].
    [ERROR] polkauction/core/route/ParachainRoutingTest has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 55.0
    [INFO] 
    [INFO] Results:
    [INFO] 
    [INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
    [INFO] 
    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD FAILURE
    [INFO] ------------------------------------------------------------------------
    [INFO] Total time:  1.343 s
    [INFO] Finished at: 2022-03-11T18:21:29+01:00
    [INFO] ------------------------------------------------------------------------
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test (default-test) on project polk-auction.core: There are test failures.
    [ERROR] 
    [ERROR] Please refer to /home/seraya/repos/polk-auction/polk-auction-core/target/surefire-reports for the individual test results.
    [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    [ERROR] There was an error in the forked process
    [ERROR] polkauction/core/route/ParachainRoutingTest has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 55.0
    [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
    [ERROR] polkauction/core/route/ParachainRoutingTest has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 55.0
    [ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:733)
    [ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:305)
    [ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:265)
    [ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1314)
    [ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1159)
    [ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:932)
    [ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
    [ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
    [ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
    [ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
    [ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
    [ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
    [ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
    [ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
    [ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
    [ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
    [ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
    [ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
    [ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
    [ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
    [ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
    [ERROR] 
    [ERROR] -> [Help 1]
    [ERROR] 
    [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
    [ERROR] Re-run Maven using the -X switch to enable full debug logging.
    [ERROR] 
    [ERROR] For more information about the errors and possible solutions, please read the following articles:
    [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
    ```

2. I still don't understand, why you decided to use different IPs for the containers. Usually it's sufficient to use different ports. For example, is there a reason for using `127.0.33.1:3308` for the db instead of `127.0.0.1:3308`?

3. Unfortunately, I'm still not able to access `polk-auction-core` on `8080`. For all other containers this works when accessing them on the exposed ports. I also checked if there is another service running on that port which was not the case:

    ```bash
    $ sudo netstat -tulpen
    Active Internet connections (only servers)
    Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode      PID/Program name       
    tcp        0      0 127.0.33.1:3308         0.0.0.0:*               LISTEN      0          39458568   2733217/docker-prox 
    tcp        0      0 127.0.200.1:80          0.0.0.0:*               LISTEN      0          39500030   2734632/docker-prox 
    tcp        0      0 127.0.100.1:8080        0.0.0.0:*               LISTEN      0          39489831   2733524/docker-prox 
    tcp        0      0 127.0.1.10:8080         0.0.0.0:*               LISTEN      0          39465617   2733030/docker-prox 
    tcp        0      0 127.0.0.10:8080         0.0.0.0:*               LISTEN      0          39471602   2732235/docker-prox
```

5. Btw, another thing that confuses me and I think I didn't mention yet: Is there a reason you added multiple `docker-compose.yml` files? Instead, you could compose them into 1 file which would also make it possible to spin up and tear down the entire infrastructure using just 1 command.
Also, while it's not a necessity, I think it would be nice to have the same prefix for all your containers:

```bash
$ docker ps
CONTAINER ID   IMAGE                                 COMMAND                  CREATED          STATUS          PORTS                                  NAMES
a79dc084fcc0   mysql:latest                          "docker-entrypoint.s…"   38 seconds ago   Up 37 seconds   33060/tcp, 127.0.33.1:3308->3306/tcp   polkauction-db
a4966ef6cefa   parity/substrate-api-sidecar:latest   "docker-entrypoint.s…"   16 minutes ago   Up 16 minutes   127.0.0.10:8080->8080/tcp              polkadot-sidecar
1693c076c96b   polk-auction-core:latest              "java -Djava.securit…"   3 hours ago      Up 6 seconds    3308/tcp, 127.0.100.1:8080->8080/tcp   polk-auction-core
f5d17313bce4   parity/substrate-api-sidecar:latest   "docker-entrypoint.s…"   3 hours ago      Up 46 seconds   127.0.1.10:8080->8080/tcp              kusama-sidecar
```

6. Thx for merging my [PR](https://github.com/CrommVardek/polk-auction-core/pull/1) so quickly:) Depending on the outcome of 5. this might be even more simplified in the future.

Please let me know if you need any further debug information, I'll be happy to assist. 👍

CrommVardek commented:
Damn my comment was not saved somehow (I posted two days ago)...

Anyway, here is what I wrote (I think):

1. It looks like you used a older Java version than version 14. If you ran it from and IDE, can you check that maven is using a JDK v14+ ? Otherwise, can you verify that your java version is 14 or higher by running `java -version` ?

2. Since some ports (8080 and 30333) are used by multiple containers, I wanted different IP (because I prefer not to change default ports). But I guess I could use the same IP for each node and its sidecar. Same for the DB and polkauction-core. I'll probably change that. I'm not too familiar with network, infrastructure and operational best practice, in my mind it made sense to have different IP (because different services).

3. What error do you have (HTTP error ? if so, which code) ? Note that accessing directly [http://127.0.100.1:8080/](http://127.0.100.1:8080/) will return a 404 http response. However you can check for example http://127.0.100.1:8080/parachain/kusama this should return 200 with a body response. 

4. No 4... :) 

5. Good idea for the prefix. I'll look into a way to have one docker-compose for everything while keeping some flexibility (running nodes or not, adding a relaychain's node and sidecar, or run only a node for specific relaychain and sidecar, etc). I'll keep you informed if I change the docker-compose files regarding this point :) 

6. Np, in the end, the linux/macOS syntax is also compatible with Windows, so I've simplified it and keep your addition and remove the previous commands.

takahser commented:
@CrommVardek Oops, that's very unfortunate.

1. You're right, I was using v11:

    ```bash
    $ java -version
    openjdk version "11.0.14" 2022-01-18
    OpenJDK Runtime Environment (build 11.0.14+9-Ubuntu-0ubuntu2.20.04)
    OpenJDK 64-Bit Server VM (build 11.0.14+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)
    ```

    I've updated it to v13:

    ```bash
    $ java --version
    openjdk 13.0.7 2021-04-20
    OpenJDK Runtime Environment (build 13.0.7+5-Ubuntu-0ubuntu120.04)
    OpenJDK 64-Bit Server VM (build 13.0.7+5-Ubuntu-0ubuntu120.04, mixed mode, sharing)

    $ mvn --version
    Apache Maven 3.6.3
    Maven home: /usr/share/maven
    Java version: 13.0.7, vendor: Private Build, runtime: /usr/lib/jvm/java-13-openjdk-amd64
    Default locale: en_US, platform encoding: UTF-8
    OS name: "linux", version: "5.4.0-100-generic", arch: "amd64", family: "unix"
    ```

    However, tests still fail:
    ```bash
    $ mvn test
    WARNING: An illegal reflective access operation has occurred
    WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
    WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
    WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
    WARNING: All illegal access operations will be denied in a future release
    [INFO] Scanning for projects...
    [WARNING] 
    [WARNING] Some problems were encountered while building the effective model for vardek:polk-auction.core:jar:0.2.0-BETA
    [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 207, column 21
    [WARNING] 
    [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
    [WARNING] 
    [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
    [WARNING] 
    [INFO] 
    [INFO] ----------------------< vardek:polk-auction.core >----------------------
    [INFO] Building polk-auction.core 0.2.0-BETA
    [INFO] --------------------------------[ jar ]---------------------------------
    [INFO] 
    [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ polk-auction.core ---
    [INFO] Using 'UTF-8' encoding to copy filtered resources.
    [INFO] Copying 6 resources
    [INFO] 
    [INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ polk-auction.core ---
    [INFO] Nothing to compile - all classes are up to date
    [INFO] 
    [INFO] --- kotlin-maven-plugin:1.4.32:compile (compile) @ polk-auction.core ---
    [WARNING] Using experimental Kotlin incremental compilation
    [INFO] Compiled 0 Kotlin files using incremental compiler
    [INFO] 
    [INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ polk-auction.core ---
    [INFO] Using 'UTF-8' encoding to copy filtered resources.
    [INFO] Copying 4 resources
    [INFO] 
    [INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ polk-auction.core ---
    [INFO] Nothing to compile - all classes are up to date
    [INFO] 
    [INFO] --- kotlin-maven-plugin:1.4.32:test-compile (test-compile) @ polk-auction.core ---
    [WARNING] Using experimental Kotlin incremental compilation
    [INFO] Compiled 0 Kotlin files using incremental compiler
    [INFO] 
    [INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ polk-auction.core ---
    [INFO] 
    [INFO] -------------------------------------------------------
    [INFO]  T E S T S
    [INFO] -------------------------------------------------------
    [ERROR] Mar 14, 2022 12:43:53 PM org.junit.platform.launcher.core.LauncherConfigurationParameters loadClasspathResource
    [ERROR] INFO: Loading JUnit Platform configuration parameters from classpath resource [file:/home/xxx/repos/polk-auction/polk-auction-core/target/test-classes/junit-platform.properties].
    [ERROR] polkauction/core/route/ParachainRoutingTest has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 57.0
    [INFO] 
    [INFO] Results:
    [INFO] 
    [INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
    [INFO] 
    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD FAILURE
    [INFO] ------------------------------------------------------------------------
    [INFO] Total time:  1.389 s
    [INFO] Finished at: 2022-03-14T12:43:53+01:00
    [INFO] ------------------------------------------------------------------------
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test (default-test) on project polk-auction.core: There are test failures.
    [ERROR] 
    [ERROR] Please refer to /home/xxx/repos/polk-auction/polk-auction-core/target/surefire-reports for the individual test results.
    [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    [ERROR] There was an error in the forked process
    [ERROR] polkauction/core/route/ParachainRoutingTest has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 57.0
    [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
    [ERROR] polkauction/core/route/ParachainRoutingTest has been compiled by a more recent version of the Java Runtime (class file version 58.0), this version of the Java Runtime only recognizes class file versions up to 57.0
    [ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:733)
    [ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:305)
    [ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:265)
    [ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1314)
    [ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1159)
    [ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:932)
    [ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
    [ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
    [ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
    [ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
    [ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
    [ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
    [ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
    [ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
    [ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
    [ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
    [ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
    [ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
    [ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
    [ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
    [ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:567)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
    [ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
    [ERROR] 
    [ERROR] -> [Help 1]
    [ERROR] 
    [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
    [ERROR] Re-run Maven using the -X switch to enable full debug logging.
    [ERROR] 
    [ERROR] For more information about the errors and possible solutions, please read the following articles:
    [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
    ```

2. Okay, got it. Thanks for elaborating on this.
3. Accessing http://127.0.100.1:8080/parachain/kusama worked, thanks
    <img width="551" alt="image" src="https://user-images.githubusercontent.com/5393704/158166124-018d21db-9349-42a2-996d-bbb87ab1d996.png">

    (Note: I forwarded 127.0.100.1:8080 => localhost:8081 here)

4. (Was 5.) Ok thanks, looking forward to that. Having 1 docker-compose config makes sense in most cases indeed. 👍

CrommVardek commented:
Hi @takahser ,

Thanks for the reply, 

Regarding point 1, that is because it is Java14+ that is needed ;)

takahser commented:
@CrommVardek apologies, somehow I messed up the java versions.
Anyway, I now upgraded to v17 :)

```bash
$ java --version
java 17.0.1 2021-10-19 LTS
Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)

$ mvn --version
Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0)
Maven home: /opt/maven
Java version: 17.0.1, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-17-oracle
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-100-generic", arch: "amd64", family: "unix"
```

However, the tests still fail:

```bash
$ mvn test
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------------< vardek:polk-auction.core >----------------------
[INFO] Building polk-auction.core 0.2.0-BETA
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-compiler-plugin/3.10.1/maven-compiler-plugin-3.10.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-compiler-plugin/3.10.1/maven-compiler-plugin-3.10.1.pom (13 kB at 89 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/34/maven-plugins-34.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/34/maven-plugins-34.pom (11 kB at 712 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-compiler-plugin/3.10.1/maven-compiler-plugin-3.10.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-compiler-plugin/3.10.1/maven-compiler-plugin-3.10.1.jar (62 kB at 2.6 MB/s)
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ polk-auction.core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 6 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.10.1:compile (default-compile) @ polk-auction.core ---
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.3.4/maven-shared-utils-3.3.4.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.3.4/maven-shared-utils-3.3.4.pom (5.8 kB at 233 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-components/34/maven-shared-components-34.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-components/34/maven-shared-components-34.pom (5.1 kB at 340 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/commons-io/commons-io/2.6/commons-io-2.6.pom
Downloaded from central: https://repo.maven.apache.org/maven2/commons-io/commons-io/2.6/commons-io-2.6.pom (14 kB at 950 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/2.0.0/plexus-component-annotations-2.0.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/2.0.0/plexus-component-annotations-2.0.0.pom (750 B at 58 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-containers/2.0.0/plexus-containers-2.0.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-containers/2.0.0/plexus-containers-2.0.0.pom (4.8 kB at 320 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/5.1/plexus-5.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/5.1/plexus-5.1.pom (23 kB at 1.4 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-java/1.1.1/plexus-java-1.1.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-java/1.1.1/plexus-java-1.1.1.pom (5.0 kB at 311 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-languages/1.1.1/plexus-languages-1.1.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-languages/1.1.1/plexus-languages-1.1.1.pom (4.6 kB at 149 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/8/plexus-8.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus/8/plexus-8.pom (25 kB at 1.6 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/ow2/asm/asm/9.2/asm-9.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/ow2/asm/asm/9.2/asm-9.2.pom (2.4 kB at 182 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/2.0.1/qdox-2.0.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/2.0.1/qdox-2.0.1.pom (16 kB at 1.1 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-api/2.11.1/plexus-compiler-api-2.11.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-api/2.11.1/plexus-compiler-api-2.11.1.pom (1.1 kB at 70 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler/2.11.1/plexus-compiler-2.11.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler/2.11.1/plexus-compiler-2.11.1.pom (8.1 kB at 541 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-components/6.6/plexus-components-6.6.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-components/6.6/plexus-components-6.6.pom (2.7 kB at 178 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/junit/junit-bom/5.8.2/junit-bom-5.8.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/junit/junit-bom/5.8.2/junit-bom-5.8.2.pom (5.6 kB at 402 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.4.1/plexus-utils-3.4.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.4.1/plexus-utils-3.4.1.pom (8.0 kB at 613 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-manager/2.11.1/plexus-compiler-manager-2.11.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-manager/2.11.1/plexus-compiler-manager-2.11.1.pom (1.1 kB at 88 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-javac/2.11.1/plexus-compiler-javac-2.11.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-javac/2.11.1/plexus-compiler-javac-2.11.1.pom (1.2 kB at 86 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compilers/2.11.1/plexus-compilers-2.11.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compilers/2.11.1/plexus-compilers-2.11.1.pom (1.3 kB at 94 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.3.4/maven-shared-utils-3.3.4.jar
Downloading from central: https://repo.maven.apache.org/maven2/commons-io/commons-io/2.6/commons-io-2.6.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/2.0.0/plexus-component-annotations-2.0.0.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-java/1.1.1/plexus-java-1.1.1.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/ow2/asm/asm/9.2/asm-9.2.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-shared-utils/3.3.4/maven-shared-utils-3.3.4.jar (153 kB at 5.9 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/2.0.1/qdox-2.0.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-component-annotations/2.0.0/plexus-component-annotations-2.0.0.jar (4.2 kB at 101 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-api/2.11.1/plexus-compiler-api-2.11.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-java/1.1.1/plexus-java-1.1.1.jar (54 kB at 1.0 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.4.1/plexus-utils-3.4.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/ow2/asm/asm/9.2/asm-9.2.jar (122 kB at 2.3 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-manager/2.11.1/plexus-compiler-manager-2.11.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/commons-io/commons-io/2.6/commons-io-2.6.jar (215 kB at 3.8 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-javac/2.11.1/plexus-compiler-javac-2.11.1.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-api/2.11.1/plexus-compiler-api-2.11.1.jar (27 kB at 457 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/2.0.1/qdox-2.0.1.jar (334 kB at 5.5 MB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-manager/2.11.1/plexus-compiler-manager-2.11.1.jar (4.7 kB at 68 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-javac/2.11.1/plexus-compiler-javac-2.11.1.jar (23 kB at 322 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.4.1/plexus-utils-3.4.1.jar (264 kB at 3.4 MB/s)
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- kotlin-maven-plugin:1.4.32:compile (compile) @ polk-auction.core ---
[WARNING] Using experimental Kotlin incremental compilation
ERROR: symbolic reference class is not accessible: class sun.nio.ch.FileChannelImpl, from class com.intellij.util.io.FileChannelUtil (unnamed module @76216830)
java.lang.IllegalAccessException: symbolic reference class is not accessible: class sun.nio.ch.FileChannelImpl, from class com.intellij.util.io.FileChannelUtil (unnamed module @76216830)
        at java.base/java.lang.invoke.MemberName.makeAccessException(MemberName.java:955)
        at java.base/java.lang.invoke.MethodHandles$Lookup.checkSymbolicClass(MethodHandles.java:3686)
        at java.base/java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:3646)
        at java.base/java.lang.invoke.MethodHandles$Lookup.findVirtual(MethodHandles.java:2680)
        at com.intellij.util.io.FileChannelUtil.setupUnInterruptibleHandle(FileChannelUtil.java:36)
        at com.intellij.util.io.FileChannelUtil.<clinit>(FileChannelUtil.java:18)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper$FileContext$1.execute(ReadWriteDirectBufferWrapper.java:59)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper$FileContext$1.execute(ReadWriteDirectBufferWrapper.java:49)
        at com.intellij.openapi.util.io.FileUtilRt.doIOOperation(FileUtilRt.java:957)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper$FileContext.<init>(ReadWriteDirectBufferWrapper.java:49)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper.create(ReadWriteDirectBufferWrapper.java:35)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
[INFO] Rebuilding because of possible caches corruption: com.intellij.util.io.PersistentEnumeratorBase$CorruptedException: PersistentEnumerator storage corrupted /home/xxx/repos/polk-auction/polk-auction-core/target/kotlin-ic/compile/caches-jvm/inputs/source-snapshot.tab
[INFO] Non-incremental compilation will be performed: CACHE_CORRUPTION
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: Could not initialize class com.intellij.util.io.FileChannelUtil
        at com.intellij.util.io.DirectBufferWrapper.doCreate(DirectBufferWrapper.java:61)
        at com.intellij.util.io.DirectBufferWrapper.getBuffer(DirectBufferWrapper.java:38)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper.getBuffer(ReadWriteDirectBufferWrapper.java:22)
        at com.intellij.util.io.StorageLock.createValue(StorageLock.java:229)
        at com.intellij.util.io.StorageLock.get(StorageLock.java:154)
        at com.intellij.util.io.PagedFileStorage.getBufferWrapper(PagedFileStorage.java:324)
        at com.intellij.util.io.PagedFileStorage.getBuffer(PagedFileStorage.java:303)
        at com.intellij.util.io.PagedFileStorage.put(PagedFileStorage.java:206)
        at com.intellij.util.io.PagedFileStorage.fillWithZeros(PagedFileStorage.java:279)
        at com.intellij.util.io.PagedFileStorage.resize(PagedFileStorage.java:253)
        at com.intellij.util.io.ResizeableMappedFile.expand(ResizeableMappedFile.java:105)
        at com.intellij.util.io.ResizeableMappedFile.ensureSize(ResizeableMappedFile.java:74)
        at com.intellij.util.io.ResizeableMappedFile.putInt(ResizeableMappedFile.java:207)
        at com.intellij.util.io.PersistentEnumeratorBase.markDirty(PersistentEnumeratorBase.java:619)
        at com.intellij.util.io.PersistentEnumeratorBase.<init>(PersistentEnumeratorBase.java:192)
        at com.intellij.util.io.PersistentBTreeEnumerator.<init>(PersistentBTreeEnumerator.java:73)
        at com.intellij.util.io.PersistentEnumeratorDelegate.createDefaultEnumerator(PersistentEnumeratorDelegate.java:69)
        at com.intellij.util.io.PersistentHashMap.<init>(PersistentHashMap.java:165)
        at com.intellij.util.io.PersistentHashMap.<init>(PersistentHashMap.java:150)
        at com.intellij.util.io.PersistentHashMap.<init>(PersistentHashMap.java:141)
        at com.intellij.util.io.PersistentHashMap.<init>(PersistentHashMap.java:133)
        at com.intellij.util.io.PersistentHashMap.<init>(PersistentHashMap.java:120)
        at org.jetbrains.kotlin.incremental.storage.CachingLazyStorage.createMap(CachingLazyStorage.kt:109)
        at org.jetbrains.kotlin.incremental.storage.CachingLazyStorage.getStorageOrCreateNew(CachingLazyStorage.kt:52)
        at org.jetbrains.kotlin.incremental.storage.CachingLazyStorage.set(CachingLazyStorage.kt:68)
        at org.jetbrains.kotlin.incremental.snapshots.FileSnapshotMap.compareAndUpdate(FileSnapshotMap.kt:50)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileImpl$rebuild(IncrementalCompilerRunner.kt:96)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileImpl(IncrementalCompilerRunner.kt:124)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compile(IncrementalCompilerRunner.kt:74)
        at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compile$default(IncrementalCompilerRunner.kt:65)
        at org.jetbrains.kotlin.incremental.IncrementalJvmCompilerRunnerKt.makeIncrementally(IncrementalJvmCompilerRunner.kt:84)
        at org.jetbrains.kotlin.maven.K2JVMCompileMojo.runIncrementalCompiler(K2JVMCompileMojo.java:270)
        at org.jetbrains.kotlin.maven.K2JVMCompileMojo.execCompiler(K2JVMCompileMojo.java:234)
        at org.jetbrains.kotlin.maven.K2JVMCompileMojo.execCompiler(K2JVMCompileMojo.java:55)
        at org.jetbrains.kotlin.maven.KotlinCompileMojoBase.execute(KotlinCompileMojoBase.java:209)
        at org.jetbrains.kotlin.maven.K2JVMCompileMojo.execute(K2JVMCompileMojo.java:222)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:301)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:211)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:157)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:121)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:127)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: Could not initialize class com.intellij.util.io.FileChannelUtil
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at com.intellij.util.io.DirectBufferWrapper.doCreate(DirectBufferWrapper.java:47)
        ... 58 more
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.intellij.util.io.FileChannelUtil
        at com.intellij.util.io.ReadWriteDirectBufferWrapper$FileContext$1.execute(ReadWriteDirectBufferWrapper.java:59)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper$FileContext$1.execute(ReadWriteDirectBufferWrapper.java:49)
        at com.intellij.openapi.util.io.FileUtilRt.doIOOperation(FileUtilRt.java:957)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper$FileContext.<init>(ReadWriteDirectBufferWrapper.java:49)
        at com.intellij.util.io.ReadWriteDirectBufferWrapper.create(ReadWriteDirectBufferWrapper.java:35)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.526 s
[INFO] Finished at: 2022-03-15T11:09:36+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.4.32:compile (compile) on project polk-auction.core: Compilation failure -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

Could you make your codebase compatible to Java v17 since [public support for openjdk 14 ended in September 2020](https://www.oracle.com/java/technologies/java-se-support-roadmap.html)?

CrommVardek commented:
Hi @takahser , I'll do that and make the project Java17 compatible. Moreover, the error is most likely linked to the JDK used. Once upgraded (and working) for Java17, I'll notify you here. I should have done this before as it was suggested to upgrade Java in the [first milestone delivery PR](https://github.com/w3f/Grant-Milestone-Delivery/pull/259)

CrommVardek commented:
Hi @takahser ,

I've updated the polkauction-core repo with upgrades of Java, Maven and Kotlin versions. Along with the dockerfile to the corresponding versions of Java and Maven.

I've verified that `mvn test`, `mvn clean install` and `docker build -t polk-auction-core .`  works properly.

Regarding `mvn test` you might come across the following logs, in the end it is not a error that affect the build nor the results of the test, so you can ignore it : 
`[ERROR] mars 16, 2022 9:58:27 PM org.junit.platform.launcher.core.LauncherConfigurationParameters loadClasspathResource
[ERROR] INFO: Loading JUnit Platform configuration parameters from classpath resource `

takahser commented:
Hi @CrommVardek 
Thanks for your patience and sorry for the delay here.
I pulled and tested your changes, now everything seems to work just fine.
Unfortunately, I didn't find an easy way to test the frontend without changing all the URIs since I was using an external server for running the code and hence had to forward all the IPs with their ports to different ports on my local machine. Naturally, since the ports aren't identical I'm not able to get any response from the services when the requests are sent by the UI. So instead, I manually sent these requests to the corresponding APIs and they all delivered seemingly useful results. 👍
Since I couldn't display them in the UI due to the reasons explained, could you provide me with 1 screenshot of each of the following pages:

- _Parachains_ page
- _Crowdloans_ page
- _Auction_ page

You can directly embed them into your comment here. After that, I'm ready to approve your delivery. 👍

CrommVardek commented:
Hi @takahser ,

No problem. I'm not really knowledgeable when it comes to network redirection. But could changing the api url in the [.env.prod file](https://github.com/CrommVardek/polk-auction-ui/blob/develop/.env.prod) help ? Because I find it a little sad you went into so much trouble with this delivery and yet not be able to view the end result.

Anyway, here are a few screenshots :

![KusamaParachainsOnHoverLeasePeriods](https://user-images.githubusercontent.com/14314577/159745127-bf6cf57d-5d47-44d1-a267-9272d2261730.JPG)
![KusamaParachains](https://user-images.githubusercontent.com/14314577/159745130-8eb09cde-4cc1-4108-ba7d-0a1b7c69e953.JPG)
![KusamaCrowdloans](https://user-images.githubusercontent.com/14314577/159745133-c00dac22-8a99-45d4-9755-e6c5b3c4898b.JPG)
![KusamaAuction](https://user-images.githubusercontent.com/14314577/159745138-a505628b-eb8e-4323-9282-8a47a9e91e1c.JPG)
![PolkadotNoAuction](https://user-images.githubusercontent.com/14314577/159745144-5fb90f74-c591-424f-a9df-31583dbe3477.JPG)
![PolkadotCrowdloan](https://user-images.githubusercontent.com/14314577/159745145-aadd4e23-15f2-4e61-9b24-3c8a675dc5fb.JPG)
![PolkadotParachains](https://user-images.githubusercontent.com/14314577/159745158-8c633a8f-9c8a-47a2-9784-859baefb4fb4.JPG)

takahser reviewed (approved):
Hey @CrommVardek
Thanks for the screenshots and the hint to change the endpoint in the env file. That actually helped, I'm now able to view all the pages:
![image](https://user-images.githubusercontent.com/5393704/159900294-f699ef4f-a00f-4231-9dad-d47cbf82e70a.png)

I accepted your delivery, thanks for the hard work you put into it. You can find my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/55ba8e61290fe503e32bef5433e47d04141bc18f/evaluations/polk-auction_2_takahser.md). Looking forward to M3! 🙌

takahser commented:
@CrommVardek one more thing, I was told that the billing address is missing on your invoice:

> Web 3.0 Technologies Foundation
> Baarerstrasse 14
> 6300 Zug
> Switzerland
> 

May I ask you to add it to your invoice, so it can be processed?

CrommVardek commented:
@takahser Thanks ! I will take into account your general feedback you added on the evaluation and I'll make changes as discussed. I'll try to do it before the third milestone.

I've just added the billing address and re-upload the invoice - sorry I re-used the template from the 1st milestone without checking the new requirements.

takahser commented:
@CrommVardek that's not a problem at all, thanks for re-uploading the invoice. I've just forwarded it, please allow for up to 2 weeks until the payment arrives.
